config/config.go

```diff
@@ -35,18 +37,58 @@ var (
 	ErrRemoteConfigNotFound  = errors.New("remote config not found")
 	ErrRemoteConfigEmptyURL  = errors.New("remote config: empty URL")
 	ErrRemoteConfigEmptyName = errors.New("remote config: empty name")
+
+	// ActiveFS provides a configurable entry point to the billy.Filesystem in active use for reading global and
+	// system level configuration; override in a test to use a fake filesystem then reset to config.DefaultFS() to
+	// restore the default behavior.
+	ActiveFS = DefaultFS()
```

Comment:
If this is for testing purposes only, I would make it private instead.

From a public API perspective, we probably want to get `Config` to work more like the core functionality, whereby such configuration is provided via `...WithOptions` when doing New/Load configs.

Reply:
While looking at this I see that a `Repository` stores a `billy.Filesystem` rooted at the worktree of the repo, which does not allow access to files outside the repo.

The core of #395 is that by default `git` operations require files outside the worktree as the default behavior is to merge system and global config into local config before executing any operation.

I can look into adding a `...WithOptions` parameter to pass in the root file system as well as the worktree fs though this may get confusing to consumers, passing in two different filesystems, especially as one is most often a subset of the other.